### DISCUSSION TITLE
Score output

### DISCUSSION THREAD

**Discussion by LightningThunder1** [Disservin/fastchess#706] [Ideas]
Hi, I was wondering if it was possible to add a feature from cutechess?

Examples are for the End of Tournament stats:

Engine 1 playing White: Wins/Draws/Losses - 0.500.
Engine 1 playing Black: Wins/Draws/Losses - 0.500.
White vs Black: Wins/Draws/Losses - 0.500.

Total wins by adjudication, total draws by adjudication, by white/black as well etc.

This is a long shot, it won't happen but I'll pointlessly try.
New variant - 3check?

**Comment by gahtan-syarif** [comment]
Refer to https://github.com/Disservin/fastchess/issues/342

**Comment by LightningThunder1** [comment]
Oh, I did read that. I'd thought I'll add a few more.

**Comment by gahtan-syarif** [comment]
as for 3check chess, im not sure but there doesnt seem to be many UCI engines that support 3check? unlike chess960

**Comment by LightningThunder1** [comment]
There is is some atleast.

I won't post this in the 'issues' because I'm not sure if this is an actual issue or that it matters.... but  LLR stays 0 when the score = 100%.

It should have stopped a while ago - If a side wins 618 games in a row, that's way beyond 95-99% Stat. signficance. 

"Results of stockfishone vs WrongEval (0.1+0.1, 1t, 16MB, UHO_Lichess_4852_v1.epd):
Elo: inf +/- nan, nElo: inf +/- nan
LOS: 100.00 %, DrawRatio: 0.00 %, PairsRatio: inf
Games: 618, Wins: 618, Losses: 0, Draws: 0, Points: 618.0 (100.00 %)
Ptnml(0-2): [0, 0, 0, 0, 309]
LLR: 0.00 (-2.94, 2.94) [0.00, 5.00]"

I had to manually stop it.

**Reply by gahtan-syarif** [reply]
The only uci engine i know of that supports 3check is fairy-stockfish, and thats because its designed form the beginning to be variant-agnostic. So its very unlikely any other chess variant will be added to fastchess unless it gains popularity/receives widespread support among chess engines like chess960 does

As for the sprt not stopping when score is 100% its already been fixed with this commit https://github.com/Disservin/fastchess/commit/8fdb0dd3b2a69d6b595bcb6ef253278db4cd36b8

**Reply by gahtan-syarif** [reply]
If you really need 3check chess support i can try to make you a custom modded version of fastchess with 3check gamemode, but of course this wont be merged to main branch

**Reply by LightningThunder1** [reply]
> If you really need 3check chess support i can try to make you a custom modded version of fastchess with 3check gamemode, but of course this wont be merged to main branch

Okay thankyou.

**Reply by gahtan-syarif** [reply]
Ill @ you when i make it, but no promises if/when it will be done

**Comment by LightningThunder1** [comment]
I only downloaded the last version that was compiled.

**Reply by gahtan-syarif** [reply]
Latest compiled dev version can be found here https://github.com/Disservin/fastchess/actions/runs/10760813430 this version already has the unending sprt issue fixed

**Reply by LightningThunder1** [reply]
> Latest compiled dev version can be found here https://github.com/Disservin/fastchess/actions/runs/10760813430 this version already has the unending sprt issue fixed

LLR is finally going upwards. I'm still surprised it's still going.

If you were to toss a 50/50 coin = 45 flips : 45 the number of heads.
The chances of that happening is 1 in 35 trillion.
I'm obviously misunderstanding the LLR/SPRT and how it works...

Results of stockfishone vs WrongEval (0.1+0.1, 1t, 16MB, UHO_Lichess_4852_v1.epd):
Elo: inf +/- nan, nElo: inf +/- nan
LOS: 100.00 %, DrawRatio: 0.00 %, PairsRatio: inf
Games: 618, Wins: 618, Losses: 0, Draws: 0, Points: 618.0 (100.00 %)
Ptnml(0-2): [0, 0, 0, 0, 309], WL/DD Ratio: -nan
LLR: 0.69 (-2.94, 2.94) [0.00, 5.00]

**Reply by gahtan-syarif** [reply]
LLR and nElo by itself is impossible to calculate when theres only one result, as it relies on the variance of the score. What i did to fix it is by regularizing the result when the result is completely one-sided. But this isnt 100% accurate so the test ending can be slow. If you want it to end faster you can set the model to logistic

**Reply by LightningThunder1** [reply]
I'm trying to get my values to match up with the ones on fishtest.

I've changed the config file to match the ptml, wins losses draws etc, but the values are somewhat different.
sprt values - elo0: 0.50  alpha: 0.05  elo1: 2.50  beta: 0.05 (normalized)

"elo 8.54 [5.37,11.71] (95%)" - i keep getting around 8.70 i think?
https://tests.stockfishchess.org/tests/live_elo/66e8ab2386d5ee47d953aba8

I'll try and modify the code and then compile it, see if i can get them to match up or not.

**Reply by gahtan-syarif** [reply]
Both those results are correct, its just that fastchess displays regular elo while fishtest displays sprt-adjusted elo.
